app/src/main/java/org/mozilla/fenix/perf/NavGraphProvider.kt

```diff
@@ -54,6 +58,11 @@ object NavGraphProvider {
             "`NavGraphProvider.inflateNavGraphAsync` to be called before this method with the same " +
             "`NavController` instance. If this occurred in a test, you probably need to add the " +
             "DisableNavGraphProviderAssertionRule.")
-        runBlockingIncrement { inflationJob.join() }
+        runBlockingIncrement {
+            if (!isGraphSet) {
```

Comment:
Rather than creating an additional boolean flag, can we query the `navController` for its state? e.g.
```kotlin
if (navController.graph == null) {
    // set the graph
}
```

Then again, this could be a bad idea: if Google ever changes `navController.graph` for an uninitialized graph, our code could break.

Reply:
So we tried that one. `NavController.graph == null` is always false since `NavController` checks for a null `graph` and if it is `null` it throws an error, which is why we went with the boolean option.